Added error handling if resumption token is not a valid base64 string

bumann-sbb opened the pull request:
When sending an invalid resumption token (not a valid base64 encoded String) the library throws an unhandled exception (`IllegalArgumentException`).
This change handles this exception and throws a `BadResumptionTokenException` and thus produces a valid OAI PMH response.

bumann-sbb committed:
Added error handling if resumption token is not a valid base64 string

bumann-sbb commented:
The build apparently does not work because I'm working on a fork.

pdurbin commented:
Yes, apparently so. I opened an issue about this:

- #273

bumann-sbb commented:
Any updates on this? Is there a different way I can contribute to avoid this error?

pdurbin commented:
@bumann-sbb no, you're doing everything right. We've just been busy! 😅 Thanks for your patience!

poikilotherm committed:
test(common): restructure tests for invalid Base64 encoded strings

- Use a parameterized test with format.parse(), not the lower level API call, making it more aligned to the pre-existing tests.
- Add a parsing test with a string that contains UTF-8 chars to ensure encoding and decoding those works and don't provoke failures.

poikilotherm committed:
doc: add release note for v5.2.2 to README

poikilotherm reviewed (approved):
Hi @bumann-sbb !

Thanks for the PR, good catch! :heart_hands: 

 I refactored the test a bit and will merge now. After I merged some of the open PRs by Renovate bot, I'm going to cut a 5.2.2 release, should be available soon. You can obviously already test using the 5.2.2-SNAPSHOT.

Cheers!

poikilotherm commented:
5.2.2 has been released, should be available soon from Central.
